server/src/main/java/org/apache/cassandra/sidecar/job/RepairJob.java

```
@@ -0,0 +1,244 @@
+                         .compose(f -> {
+                             boolean isTimeout = maxWaitTimePromise.future().result();
+                             if (isTimeout)
+                             {
+                                 LOGGER.error("Timer ran out before the repair job completed. Repair took too long");
+                                 return Future.failedFuture("Repair job taking too long");
+                             }
+                             return repairPromise.future();
+                         });
+        }
+        return repairPromise.future();
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String name()
+    {
+        return OPERATION;
+    }
+
+    private Map<String, String> generateRepairOptions(RepairPayload repairPayload)
+    {
+        Map<String, String> options = new HashMap<>();
+
+        List<String> tables = repairPayload.tables();
+        if (tables != null && !tables.isEmpty())
+        {
+            options.put(RepairOptions.COLUMNFAMILIES.getValue(), String.join(",", tables));
+        }
+
+        Boolean isPrimaryRange = repairPayload.isPrimaryRange();
+        if (isPrimaryRange != null)
+        {
+            options.put(RepairOptions.PRIMARY_RANGE.getValue(), String.valueOf(isPrimaryRange));
+        }
+        // TODO: Verify use-cases involving multiple DCs
+
+        String dc = repairPayload.datacenter();
+        if (dc != null)
+        {
+            options.put(RepairOptions.DATACENTERS.getValue(), dc);
+        }
+
+        List<String> hosts = repairPayload.hosts();
+        if (hosts != null && !hosts.isEmpty())
+        {
+            options.put(RepairOptions.HOSTS.getValue(), String.join(",", hosts));
+        }
+
+        if (repairPayload.startToken() != null && repairPayload.endToken() != null)
+        {
+            options.put(RepairOptions.RANGES.getValue(), repairPayload.startToken() + ":" + repairPayload.endToken());
+        }
```

I feel it is easy enough to also check that start and end tokens form a valid range, i.e. start <= end, in the spirit of fail early.